src/apply_conf.rs

```diff
@@ -0,0 +1,297 @@
+use std::ffi::OsStr;
+use std::fs;
+use std::io::Write;
+use std::os::unix::fs::OpenOptionsExt;
+use std::path::Path;
+
+use anyhow::{anyhow, Context};
+use log::{debug, info, warn};
+use network_interface::{NetworkInterface, NetworkInterfaceConfig};
+
+use crate::types::Host;
+use crate::HOST_MAPPING_FILE;
+
+const CONNECTION_FILE_EXT: &str = "nmconnection";
+
+pub(crate) fn apply(source_dir: &str, destination_dir: &str) -> Result<(), anyhow::Error> {
+    let hosts = parse_config(source_dir).context("Parsing config")?;
+    debug!("Loaded hosts config: {hosts:?}");
+
+    let network_interfaces = NetworkInterface::show()?;
+    debug!("Retrieved network interfaces: {network_interfaces:?}");
+
+    let host = identify_host(hosts, &network_interfaces)
+        .ok_or_else(|| anyhow!("None of the preconfigured hosts match local NICs"))?;
+    info!("Identified host: {}", host.hostname);
+
+    copy_connection_files(host, &network_interfaces, source_dir, destination_dir)
+}
+
+fn parse_config(source_dir: &str) -> Result<Vec<Host>, anyhow::Error> {
+    let config_file = Path::new(source_dir).join(HOST_MAPPING_FILE);
+
+    let file = fs::File::open(config_file)?;
+    let mut hosts: Vec<Host> = serde_yaml::from_reader(file)?;
+
+    // Ensure lower case formatting.
+    hosts.iter_mut().for_each(|h| {
+        h.interfaces
+            .iter_mut()
+            .for_each(|i| i.mac_address = i.mac_address.to_lowercase());
+    });
```

Comment:
You lack tests for this "to_lowercase" thing.

Reply:
This is more of a sanity check but you're definitely right. Added a test for it.